sky/provision/azure/instance.py

```diff
@@ -704,6 +709,30 @@ def _fetch_and_map_status(node, resource_group: str) -> None:
     return statuses
 
 
+def _get_cluster_nsg(network_client: Client, resource_group: str,
```

Comment:
Could you add # TODO: We can remove this after 0.8.0 to ignore legacy nsg names.